sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -70,11 +70,25 @@ public virtual RecordEntry FindMatch(Request request, IList<RecordEntry> entries
             int bestScore = int.MaxValue;
             RecordEntry bestScoreEntry = null;
 
+            bool isTrack1Record = entries.Any(item => !string.IsNullOrEmpty(item.EncodedRequestUri));
+
             foreach (RecordEntry entry in entries)
             {
                 int score = 0;
 
-                if (!AreUrisSame(entry.RequestUri, uri))
+                var recordRequestUri = entry.RequestUri;
+                if (isTrack1Record)
+                {
+                    //there's no domain name for request uri in track 1 record, so add it from reqeust uri
+                    int len = 8; //length of "https://"
```

Comment:
are we positive that schema is always https?

Reply:
the **len** we used as start index for searching "/" to get end index of domain name, so 8 is good here even for http://